rfd/0081/health-check-draft.md

```diff
@@ -0,0 +1,58 @@
+# Mariposa Health Check Service
+
+The Mariposa (RFD36) health check service will be responsible for monitoring the health of user services. It establishes a set of supported health checks and provides an API for defining health checks that should be ran against project services. These health check definitions will be persisted in a proper data store (Postgres/Manatee?).
+
+The Health Check service process(es) will read user health check definitions, and coordinate which health check process will be responsible for running any given definition. 
+
+Failed health checks will be broadcast to interested services, such as the Convergence service.
+
+## Health check types
+
+The available health checks include:
+
+* HTTP(s) - Test the response of an HTTP request against a given service/port. Any part of the response may be validated including response code, headers, or body.
+* TCP - Test that a service/port is accepting TCP connections.
+* ContainerPilot - Interface with ContainerPilot running within a service container and accept the health state returned by it.
+
+In addition to the health checks specified, the health check service will interface with VMAPI and ChangeFeed to understand when a container state transitions between stopped/paused/running. Stopped/paused will be considered unhealthy and the above health checks will not be attempted against containers in these states.
```

Comment:
> then it needs to poll the Convergence service for its configuration

You mean the Project service? It sounds like we're possibly in agreement (or mutual understanding or whatever) that it should poll the Project service (where health checks have been configured) and push health change events directly to the Convergence service. Convergence will not poll health check agent(s) (with you on the agent thing).

Since we're talking about many of these agents running, my last thought of health check service watching changefeed/vm api doesn't make sense. It makes sense to do that in Convergence service as initially stated, and health check agent is supplemental information to that.